### PR TITLE
fix(eks): aws-node-termination-handler incorrectly deployed to on-demand instances as well

### LINKED ISSUE #12368: (aws-eks): spot node termination handler is deployed to all nodes
When adding autoscaling group capacity the `aws-node-termination-handler` is deployed as Helm chart. There is a `nodeSelector` applied which should only deploy the `DaemonSet` to nodes labeled with `lifecycle=Ec2Spot`. Unfortunately this was not done correctly.

Current implementation:

```ts
'nodeSelector.lifecycle': LifecycleLabel.SPOT
```

When I get the Helm values (`helm get values ...`) it looks like this:

```
nodeSelector: {}
nodeSelector.lifecycle: Ec2Spot
```

Correct implementation:

```ts
nodeSelector:{
  lifecycle: LifecycleLabel.SPOT
}
```

I already created a fix. I will submit a PR.

--- 

This is :bug: Bug Report

### PR DESCRIPTION
**Pull request by dirknilius**
fixes #12368

The former implementation was using the string format for submitting Helm values in the CLI (`parent.child=value`). This does not work here as JavaScript does not expand the string. We need to define the values is the correct hierarchy.

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

**Review by iliapolo — APPROVED** [review]
Sweet, thanks!

**Comment by dirknilius** [comment]
@iliapolo Hey,

somehow the CodeBuild failed. Unfortunately I cannot download the logs. It just loads with a few kb/s and then stops. I am also not able to run the tests in Gitpod. The docs (`CONTRIBUTING.md`) states I have to run `yarn build`. But this cannot work as there is no Docker installed. Also the partial build relies on Docker. How could this work?

File: packages/@aws-cdk/aws-eks/lib/cluster.ts [hunk 1]
```diff
@@ -1450,7 +1450,9 @@ export class Cluster extends ClusterBase {
         repository: 'https://aws.github.io/eks-charts',
         namespace: 'kube-system',
         values: {
-          'nodeSelector.lifecycle': LifecycleLabel.SPOT,
+          nodeSelector:{
```
Comment (iliapolo) [inline packages/@aws-cdk/aws-eks/lib/cluster.ts:1453]:
```suggestion
          nodeSelector: {
```

File: packages/@aws-cdk/aws-eks/lib/cluster.ts [hunk 2]
```diff
@@ -1450,7 +1450,9 @@ export class Cluster extends ClusterBase {
...
+            lifecycle: LifecycleLabel.SPOT,
+          }
```
Comment (iliapolo) [inline packages/@aws-cdk/aws-eks/lib/cluster.ts:1455]:
```suggestion
          },
```

**Comment by iliapolo** [comment]
@dirknilius It was just some linting stuff:

```console
@aws-cdk/aws-cloudwatch-actions: Done in 24.53s.
@aws-cdk/aws-eks: /codebuild/output/src958570000/src/github.com/aws/aws-cdk/packages/@aws-cdk/aws-eks/lib/cluster.ts
@aws-cdk/aws-eks:   1453:24  error  Missing space before value for key 'nodeSelector'  key-spacing
@aws-cdk/aws-eks:   1455:12  error  Missing trailing comma                             comma-dangle
@aws-cdk/aws-eks: âœ– 2 problems (2 errors, 0 warnings)
@aws-cdk/aws-eks:   2 errors and 0 warnings potentially fixable with the `--fix` option.
@aws-cdk/aws-eks: Error: /codebuild/output/src958570000/src/github.com/aws/aws-cdk/node_modules/esli
```

I got this from the `Build Logs` link of the `CodeBuild` report, not sure why it didn't open for you...as for the gitpod and partial build, i'll take a look.

I took the liberty and just committed a fix from the review UI, so hopefully it will get merged soon.

Thanks!

**Review by iliapolo — APPROVED** [review]

**Pull request merged**